### PR TITLE
tidy up around how GitStore stores remotes

### PR DESCRIPTION
**Pull request by shiftkey**
Some tidy up I did while investigating #5936, to better understand this part of the codebase.

No functional changes, but:

 - better jsdoc statements for `GitStore.*remote*` properties
 - more descriptive names for `GitStore.remote` and `GitStore.upstream`
 - additional test to confirm remotes are emitted in alphabetical order from Git
 - updated `remote-test.ts` to change from `chai` to jest builtins

### CODE REVIEW

File: app/src/lib/stores/git-store.ts [hunk 1]
```diff
@@ -344,11 +344,11 @@ export class GitStore extends BaseStore {
       return gitHubRepository.defaultBranch
     }
 
-    if (this.remote != null) {
+    if (this.currentRemote != null) {
```
Comment (iAmWillShepherd) [inline app/src/lib/stores/git-store.ts]:
```suggestion
    if (this.currentRemote !== null) {
```

File: app/src/lib/stores/git-store.ts [hunk 2]
```diff
@@ -766,8 +766,8 @@ export class GitStore extends BaseStore {
     const remotes = new Map<string, IRemote>()
 
     // We want to fetch the current remote first
-    if (this.remote) {
-      remotes.set(this.remote.name, this.remote)
+    if (this.currentRemote) {
```
Comment (iAmWillShepherd) [inline app/src/lib/stores/git-store.ts]:
```suggestion
    if (this.currentRemote !== null) {
```

File: app/src/lib/stores/git-store.ts [hunk 3]
```diff
@@ -776,8 +776,8 @@ export class GitStore extends BaseStore {
     }
 
     // And finally the upstream if we're a fork
-    if (this.upstream) {
-      remotes.set(this.upstream.name, this.upstream)
+    if (this.upstreamRemote) {
```
Comment (iAmWillShepherd) [inline app/src/lib/stores/git-store.ts]:
```suggestion
    if (this.upstreamRemote !== null) {
```

**Comment by shiftkey** [comment]
:apple:

**Comment by shiftkey** [comment]
🍒

**Pull request merged**